src/base/Gas.ml

```diff
@@ -104,13 +104,31 @@ module ScillaGas (SR : Rep) (ER : Rep) = struct
   (* this is a dynamic cost. *)
 
   let stmt_cost scon =
+    let map_sort_cost l =
+      match l with
+      | Map ((kt, vt), kvlist) ->
+          let len = Caml.Hashtbl.length kvlist in
+          let depth = map_depth (MapType (kt, vt)) in
+          if len > 0 && depth > 0 then
+            let log_len = Int.of_float (Float.log (Int.to_float len)) in
+            len * log_len * depth
```

Comment:
Sorry, but what's the logic here? Is it something like n*log(n) for sorting one layer of the map, and then multiplied by the number of layers? Because that doesn't look sound - what if the top layer is just one entry, but there is a massive map in the next layer?